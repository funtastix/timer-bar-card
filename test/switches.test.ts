import { multiply, HomeAssistant, PlaywrightBrowser, PlaywrightElement } from "hass-taste-test";
import { toMatchDualSnapshot, waitForTimerTime } from "./util";

expect.extend({ toMatchDualSnapshot });

const CONFIGURATION_YAML = `
input_number:
  slider:
    min: 0
    max: 10
    initial: 5
input_boolean:
  ${multiply(2, (i) => `
  switch${i}:
`)}
`;

let hass: HomeAssistant<PlaywrightElement>;

beforeAll(async () => {
  hass = await HomeAssistant.create(CONFIGURATION_YAML, {
    browser: new PlaywrightBrowser(process.env.BROWSER || "firefox"),
  });
  await hass.addResource(__dirname + "/../dist/timer-bar-card.js", "module");
}, 30000);
afterAll(async () => await hass.close());

it("Switch with fixed duration", async () => {
  const dashboard = await hass.Dashboard([
    {
      type: "custom:timer-bar-card",
      entities: ["input_boolean.switch1"],
      duration: { fixed: "00:01:00" },
    },
  ]);
  const card = dashboard.cards[0];
  await hass.callService('homeassistant', 'turn_on', {}, { entity_id: "input_boolean.switch1" });
  await waitForTimerTime(card, "00:00:58");
  await expect(card).toMatchDualSnapshot("running");
});

it("Switch with input_number duration", async () => {
  const dashboard = await hass.Dashboard([
    {
      name: "Time is 4:59",
      type: "custom:timer-bar-card",
      entities: ["input_boolean.switch2"],
      duration: { entity: "input_number.slider", "units": "minutes" },
    },
    {
      name: "Time is 1:57",
      type: "custom:timer-bar-card",
      entities: ["input_boolean.switch2"],
      duration: { entity: "input_number.slider", "units": "minutes" },
    },
  ]);
  const card = dashboard.cards[0];
  await hass.callService('homeassistant', 'turn_on', {}, { entity_id: "input_boolean.switch2" });
  await waitForTimerTime(card, "00:04:59");
  await expect(card).toMatchDualSnapshot("5-minutes");
  await hass.callService('input_number', 'set_value', { value: 2 }, { entity_id: "input_number.slider" });
  await waitForTimerTime(dashboard.cards[1], "00:01:57");
  await expect(dashboard.cards[1]).toMatchDualSnapshot("2-minutes");
});
