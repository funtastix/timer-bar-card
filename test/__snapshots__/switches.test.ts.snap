// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch with fixed duration: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"on\\" icon=\\"hass:toggle-switch-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            switch1
          </div>
          <slot></slot>
          <div class=\\"bar-container\\">
            <div class=\\"bar\\">
              <div>
              </div>
            </div>
          </div>
          <div class=\\"text-content\\">
            58
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Switch with input_number duration: 2-minutes 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <h1 class=\\"card-header\\">
      Time is 1:57
    </h1>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"on\\" icon=\\"hass:toggle-switch-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            switch2
          </div>
          <slot></slot>
          <div class=\\"bar-container\\">
            <div class=\\"bar\\">
              <div>
              </div>
            </div>
          </div>
          <div class=\\"text-content\\">
            1:57
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Switch with input_number duration: 5-minutes 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <h1 class=\\"card-header\\">
      Time is 4:59
    </h1>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"on\\" icon=\\"hass:toggle-switch-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            switch2
          </div>
          <slot></slot>
          <div class=\\"bar-container\\">
            <div class=\\"bar\\">
              <div>
              </div>
            </div>
          </div>
          <div class=\\"text-content\\">
            4:59
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;
